.github/workflows/build_aarch64.yml

```diff
@@ -0,0 +1,315 @@
+#
+# Copyright (c) 2014-present, The osquery authors
+#
+# This source code is licensed as defined by the LICENSE file found in the
+# root directory of this source tree.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+#
+
+# Due to a limitation in how GitHub Actions works, we can't reference
+# jobs in another file inside the `needs` statement.
+#
+# This configuration file takes care of Linux on aarch64.
+name: build_aarch64
+
+on:
+  push:
+    branches:
+      - 'master'
+
+    tags:
+      - '*'
+
+# If the initial code sanity checks are passing, then one job
+# per [`platform` * `build_type`] will start, building osquery
+# and generating packages that are later attached to the commit
+# (or PR) as build artifacts.
+jobs:
+  # Starts optional self-hosted runners.
+  start_aarch64_ec2_runners:
```

Comment:
At a theory level, is there a race condition here? This starts a runner and register it. In theory, and PR could pick it up? The window is pretty small though

Reply:
I don't understand the race condition you mention? What do you mean by "PR could pick it up"?

Reply:
(It's a bit far fetched, I don't think it's a blocker)

I think that this:
1. registers a worker. 
2. Said worker can then be picked up by any job with the appropriate tags. 

There is a gap between those steps were another PR could, conceivably, run on these hosts.